Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -21,55 +21,71 @@ extension OfflineMapAreasView {
     @MainActor
     class MapViewModel: ObservableObject {
         /// The portal item ID of the web map.
-        private let portalItemID: PortalItem.ID?
+        private let portalItemID: PortalItem.ID
         
         /// The offline map task.
         private let offlineMapTask: OfflineMapTask
         
         /// The preplanned map information.
         @Published private(set) var preplannedMapModels: Result<[PreplannedMapModel], Error>?
         
-        /// The offline preplanned map information sourced from downloaded mobile map packages.
-        @Published private(set) var offlinePreplannedMapModels: [PreplannedMapModel]?
+        /// A Boolean value indicating if only offline models are being shown.
+        @Published private(set) var isShowingOnlyOfflineModels = false
+        
+        /// The online map.
+        private let onlineMap: Map
         
         /// Creates an offline map areas view model for a given web map.
         /// - Parameter map: The web map.
+        /// - Precondition: `map.item?.id` is not `nil`.
         init(map: Map) {
+            precondition(map.item?.id != nil)
```

Comment:
This should be safe right?

Reply:
Had same finding in swift/issues/6467. Per my test it should be safe - If a webmap is offline enabled, it's guaranteed to have an ID.

Reply:
Had the same finding in swift/issues/6467. Per my test it should be safe - If a webmap is offline enabled (i.e. not a purely local map), it's guaranteed to have an ID.

However, I don't think it is safe to write it this way. For example, if I simply pass in a map `Map(basemapStyle: .arcGISLightGray)`, it will crash on this line due to the map doesn't have an item or ID.

I would probably do it like this:

- Make the `portalItemID` an implicitly unwrapped optional, `private let portalItemID: PortalItem.ID!`
    - Because we know that whenever we need to access it, it doesn't make sense to be `nil`
    - A map can have `nil` item, but whichever map has a `nil` item, it doesn't support offline workflow
- Make sure no method will be called when a map's item is `nil` in UI

Reply:
Should this precondition be removed or do you want to keep it as a second fail-safe? Since there is only 1 place that calls this internal initializer and a precondition is in that place, this precondition won't be triggered?

Reply:
best to stay